.python-version

```diff
@@ -0,0 +1 @@
+3.11
```

Comment:
@hynek this is going to cause friction for the contributors using pyenv (like me) since this file is managed by the `pyenv local` command. I strongly recommend gitignoring it as it tends to get in the way.
It is meant to be controlled by the end-users and for me, it often contains a virtualenv name, maybe with a few fallbacks.
Also, by using an unversioned `3.11` w/o a patch component, it's bound to introduce surprise “command not found” whenever a contributor installs a slightly newer version (as in `pyenv install 3.11.1` followed by `pyenv install 3.11.4` months later, while working on something else).
This is why, in my opinion, it's not worth intruding into contributors' setups.

Reply:
I actually wonder which use such a file has, holding a single python version. Since it doesn't get updated automatically?

Reply:
It’s the other way round: it’s the only standard we have to set a default version that’s respected by several tools including setup-python. It’s nice to have a canonical truth _somewhere_.

I’ve seen GHA offers a way to use a different name for that file but I’m not sure if it doesn’t create more problem than it solves…

Reply:
> I actually wonder which use such a file has, holding a single python version. Since it doesn't get updated automatically?

@chrysle, if you set up your shell to use pyenv and a matching Python version or venv exists (could be several values, actually), it'll automatically make use of those versions listed in `.python-version` (or activate a venv). If there's no such version, it'll fall back to the pyenv global value. As it stands now, it effectively breaks the `pyenv-virtualenv` use case (with project-specific envs being picked up by `pyenv local`). Well, kinda. The `pyenv local` command updates the contents of `.python-version` and then, Git _sees_ the file as updated. This would likely be misleading for some people that don't know how to get Git to pretend there's no changes.